src/extensions/default/AutoUpdate/main.js

```diff
@@ -148,7 +179,8 @@ define(function (require, exports, module) {
     function postMessageToNode(messageId) {
         var msg = {
             fn: messageId,
-            args: getFunctionArgs(arguments)
+            args: getFunctionArgs(arguments),
+            requester: domainID
```

Comment:
This looks fine, but can you see if we can somehow notify the user of Update failure if the requesting window is closed. That should at least provide some info to the user rather than leave him hanging.

Reply:
This is addressed , If requesting window is closed , in Node anyone of available windows is made as requesting window